pkg/machinery/config/decoder/decoder.go

```diff
@@ -164,5 +171,97 @@ func decode(manifest *yaml.Node) (target interface{}, err error) {
 		return nil, fmt.Errorf("spec decode: %w", err)
 	}
 
+	if err = validate(target, spec); err != nil {
+		return nil, err
+	}
+
 	return target, nil
 }
+
+//nolint:gocyclo
+func validate(target interface{}, spec *yaml.Node) error {
+	node, err := encoder.NewEncoder(target).Marshal()
+	if err != nil {
+		return err
+	}
+
+	src := map[string]interface{}{}
+	dst := map[string]interface{}{}
+
+	err = spec.Decode(src)
+	if err != nil {
+		return err
+	}
+
+	err = node.Decode(dst)
+	if err != nil {
+		return err
+	}
+
+	var errs *multierror.Error
+
+	keys := []string{}
+
+	var checkExtra func(interface{}, interface{})
+
+	checkExtra = func(left interface{}, right interface{}) {
+		switch v := left.(type) {
+		case map[string]interface{}:
+			r, ok := right.(map[string]interface{})
+			if !ok {
+				errs = multierror.Append(errs, fmt.Errorf("type mismatch %v, expected map", strings.Join(keys, ".")))
+
+				return
+			}
+
+			for key, value := range v {
+				if _, ok := r[key]; !ok {
+					errs = multierror.Append(errs, fmt.Errorf("extra key %v", strings.Join(append(keys, key), ".")))
```

Comment:
this message might be hard to understand for the user, `unknown field %v` ? 

as an option, what I thought about we could also print excerpt of the YAML config which wasn't parsed properly, like eventually delete all the keys which are known and what is left should be something which is not supported/known, so we marshal it back to YAML for the user as an error?

Reply:
So I've got the following output:

```
unknown keys found during decoding:
slice:
    - fields: here
      more: extra
      not: working
```

Reply:
:heart: